contracts/strategies/ape-stake/ApeStakeStrategy.sol

```diff
@@ -0,0 +1,152 @@
+pragma solidity 0.5.16;
+pragma experimental ABIEncoderV2;
+
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import "../../base/interface/IStrategy.sol";
+import "../../base/upgradability/BaseUpgradeableStrategy.sol";
+
+import "./interfaces/IApeCoinStaking.sol";
+
+contract ApeStakeStrategy is IStrategy, BaseUpgradeableStrategy {
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+
+    address public constant APE = address(0x4d224452801ACEd8B2F0aebE155379bb5D594381);
+
+    constructor() public BaseUpgradeableStrategy() {
+    }
+
+    /* ========== Initialize ========== */
+
+    function initializeBaseStrategy(
+        //  "__" for storage because we shadow _storage from GovernableInit
+        address __storage,
+        address _vault,
+        address _rewardPool
+    ) public initializer {
+        BaseUpgradeableStrategy.initialize({
+            _storage: __storage,
+            _underlying: APE,
+            _vault: _vault,
+            _rewardPool: _rewardPool,
+            _rewardToken: APE,
+            _profitSharingNumerator: 300,
+            _profitSharingDenominator: 1000,
+            _sell: true,
+            _sellFloor: 0,
+            _implementationChangeDelay: 12 hours
+        });
+    }
+
+    /* ========== View ========== */
+
+    function depositArbCheck() public view returns (bool) {
+        return true;
+    }
+
+    function unsalvagableTokens(address token) public view returns (bool) {
+        return (token == rewardToken() || token == underlying());
+    }
+
+    function investedUnderlyingBalance() public view returns (uint256) {
+        return _rewardPoolBalance().add(IERC20(underlying()).balanceOf(address(this)));
+    }
+
+    /* ========== Internal View ========== */
+
+    function _rewardPoolBalance() internal view returns (uint256) {
+        IApeCoinStaking.Position memory position = IApeCoinStaking(rewardPool()).addressPosition(address(this));
+        return position.stakedAmount;
+    }
+
+    /* ========== Internal ========== */
+
+    function _investAllUnderlying() internal {
+        uint256 underlyingBalance = IERC20(underlying()).balanceOf(address(this));
+        IERC20(underlying()).safeApprove(rewardPool(), 0);
+        IERC20(underlying()).safeApprove(rewardPool(), underlyingBalance);
+        IApeCoinStaking(rewardPool()).depositSelfApeCoin(underlyingBalance);
+    }
+
+    function _claimRewards() internal {
+        IApeCoinStaking(rewardPool()).claimSelfApeCoin();
+    }
+
+    function _liquidateReward() internal {
+        uint256 rewardBalance = IERC20(rewardToken()).balanceOf(address(this));
```

Comment:
We need additional logic to separate rewards from deposits, as underlying == rewardToken. See an example in the YEL strategy: https://github.com/harvest-finance/harvest-strategy/blob/master/contracts/strategies/yel/YelStrategy.sol#L136

Reply:
I noticed this problem and handled it differently in `doHardWork` function. But this approach is more obvious.

Reply:
I will update it

Reply:
Gotcha. Thanks for your clarification